### PR TITLE
Empty Prometheus cache before each test case

### PR DESCRIPTION
**Pull request by nan-yu**
b/281576844
b/281508753
b/281432500

### CODE REVIEW

File: e2e/nomostest/nt.go [hunk 1]
```diff
@@ -711,6 +713,26 @@ func (nt *NT) portForwardPrometheus() {
...
+}
+
+// emptyPrometheusCache empty the cache in Prometheus.
+func (nt *NT) emptyPrometheusCache() error {
+	nt.T.Helper()
+	ctx, cancel := context.WithCancel(nt.Context)
+	defer cancel()
+
+	c, err := prometheusapi.NewClient(prometheusapi.Config{
+		Address: fmt.Sprintf("http://localhost:%d", prometheusServerPort),
```
Comment (sdowell) [inline e2e/nomostest/nt.go:729]: This is using the port forward on localhost, so you'll need to get the port from `nt.prometheusPortForwarder.LocalPort()` (and you'll probably want to wrap this in a retry loop)

Comment (nan-yu) [reply e2e/nomostest/nt.go]: Good catch. Done.

**Comment by nan-yu** [comment]
/hold

Need to verify if we can pass an empty `match` parameter to delete all

File: e2e/nomostest/nt.go [hunk 2]
```diff
@@ -711,6 +714,42 @@ func (nt *NT) portForwardPrometheus() {
...
+		c, err := prometheusapi.NewClient(prometheusapi.Config{
+			Address: fmt.Sprintf("http://localhost:%d", port),
+		})
+		if err != nil {
+			return err
+		}
+		v1api := prometheusv1.NewAPI(c)
+		// DeleteSeries only marks the time series for deletion
+		if err = v1api.DeleteSeries(ctx, nil, time.Time{}, time.Now()); err != nil {
+			return err
```
Comment (karlkfi) [inline e2e/nomostest/nt.go:743]: Add extra error message context for both of these calls to the api. The errors aren't very descriptive. ex: `server_error: server error: 500`

File: e2e/nomostest/nt.go [hunk 3]
```diff
@@ -711,6 +714,42 @@ func (nt *NT) portForwardPrometheus() {
...
+		}
+		c, err := prometheusapi.NewClient(prometheusapi.Config{
+			Address: fmt.Sprintf("http://localhost:%d", port),
+		})
+		if err != nil {
+			return err
+		}
+		v1api := prometheusv1.NewAPI(c)
+		// DeleteSeries only marks the time series for deletion
+		if err = v1api.DeleteSeries(ctx, nil, time.Time{}, time.Now()); err != nil {
```
Comment (karlkfi) [inline e2e/nomostest/nt.go:742]: I believe `delete_series` is part of the admin API that needs to be enabled with the `--web.enable-admin-api` flag on the container command. 
- https://prometheus.io/docs/operating/security/#prometheus
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testdata/prometheus/prometheus.yaml#L157

Comment (nan-yu) [reply e2e/nomostest/nt.go:742]: Yes, I have to add `--web.enable-admin-api` in the deployment. Done locally. However, we have to provide a valid matcher. 
I am trying to use cluster name as the matcher.

File: e2e/nomostest/nt.go (hunk 2 above)
Comment (nan-yu) [reply e2e/nomostest/nt.go]: done

File: e2e/nomostest/nt.go [hunk 4]
```diff
@@ -711,6 +714,45 @@ func (nt *NT) portForwardPrometheus() {
...
+			return err
+		}
+		v1api := prometheusv1.NewAPI(c)
+		// DeleteSeries only marks the time series for deletion
+		if err = v1api.DeleteSeries(ctx, []string{`{__name__=~".+"}`}, time.Time{}, time.Now()); err != nil {
+			return fmt.Errorf("failed to delete all data from Prometheus cache: %v", err)
+		}
+		// CleanTombstones remove them from the disk to clear the space instantly
+		if err = v1api.CleanTombstones(ctx); err != nil {
+			return fmt.Errorf("failed to remove tombstones from the dish: %v", err)
```
Comment (karlkfi) [inline e2e/nomostest/nt.go:747]: s/dish/disk

**Review by sdowell — APPROVED** [review]

**Comment by sdowell** [comment]
/hold
Putting a hold to address Karl's comment

**Comment by sdowell** [comment]
re-lgtm
/lgtm

Feel free to unhold when you're ready to submit

**Review by karlkfi — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**